terraform/tutorial.md

```diff
@@ -0,0 +1,351 @@
+
+### Download source code
+
+In **Cloud Shell**, issue these **commands** to download a release of the Sandbox source code and extract the archive:
+
+```bash
+cd ~
+wget https://github.com/GoogleCloudPlatform/stackdriver-sandbox/archive/next19.tar.gz
+tar -xvf next19.tar.gz
+cd stackdriver-sandbox-next19
+```
+
+### Create and configure source repository
+
+To create a Cloud Source Repository for the source code and to configure Git access, issue these commands in Cloud Shell:
+
+```bash
+gcloud source repos create google-source-captures
+git config --global user.email "user@domain.tld" # substitute with your email
+git config --global user.name "first last"       # substitute with your name
+```
+
+### Upload source code to Debugger
+
+In the Debugger home page, **copy** the command (_don't click the button!_) in the "Upload a source code capture to Google servers" box, but **don't include the `LOCAL_PATH` variable**. (You will replace this with another value before executing the command.)
+
+Paste the command into your Cloud Shell prompt and add a space and a period:
+
+```bash
+gcloud beta debug source upload --project=stackdriver-sandbox-68291054 --branch=6412930C2492B84D99F3 .
+```
+
+Enter _RETURN_ to execute the command.
+
+In the Debugger home page, click the **Select Source** button under "Upload a source code capture" option, which will then open the source code.
+
+You are now ready to debug your code!
+
+### Create a snapshot
+
+Start by using the Snapshot functionality to understand the state of your variables.  In the Source capture tree, open the **`server.js`** file under **src** > **currencyservice.** 
+
+Next, click on **line 121** to create a snapshot. in a few moments, you should see a snapshot be created, and you can view the values of all variables at that point on the right side of the screen.
+
+### Create a logpoint
+
+Switch to the **Logpoint** tab on the right side. To create the logpoint:
+
+1. Again, click on **line 121** of **`server.js`** to position the logpoint.
+1. In the **Message** field, type "testing logpoint" to set the message that will be logged.
+1. Click the **Add** button. 
+
+To see all messages that are being generated in Cloud Logging from your logpoint, click the **Logs** tab in the middle of the UI. This brings up an embedded viewer for the logs.
+
+## The Cloud observability products: Monitoring, Logging, and Error Reporting
+
+Next, learn about Cloud observability products!
+
+## Cloud Monitoring
+
+### Monitoring Overview
+
+Cloud Monitoring ([documentation](https://cloud.google.com/monitoring/docs/)) is the go-to place to grasp real-time trends of the system based on SLI/SLO. SRE team and application development team (and even business organization team) can collaborate to set up charts on the monitoring dashboard using metrics sent from the resources and the applications. 
```

Comment:
I'd like to see more links here. Maybe images too if possible

Reply:
Images are not able to render in Cloud Shell tutorials, so unfortunately I had to take out the User Guide images in the tutorial.